carbon-design-system/carbon
# fix(icons-react): update commonjs direct paths

Closes #3471 

This updates our CommonJS strategy from being a re-export to inlining the CommonJS module directly as tree-shaking/DCE does not work in the former case.

Note: this is a revert to the previous known behavior for CommonJS

#### Changelog

**New**

**Changed**

- Build step for `icons-react` now includes a step to inline CommonJS modules directly to their alias path

**Removed**

#### Testing / Reviewing

- Pull down PR
- Run `yarn link` in `@carbon/icons-react`
- Re-run `yarn build`
- Create a temporary project with Create React App in another folder
  - `npx create-react-app tmp`
- Run `yarn link @carbon/icons-react`
- Update `App.js` to the following:

<details>
<summary>Code</summary>

```js
import Add32 from '@carbon/icons-react/lib/add/32';
import React from 'react';
import './App.css';

function App() {
  return (
    <Add32 />
  );
}

export default App;
```
</details>

- Verify icon renders correctly

![Screen Shot 2019-07-19 at 4 10 32 PM](https://user-images.githubusercontent.com/3901764/61565814-c9fd6800-aa3f-11e9-8158-e239929407f0.png)

- Run `yarn build`
- Run `npx source-map-explorer build/static/js/main.XYZ`, changing `XYZ` at the end to the specific chunk hash you have from your build
  - My exact command was: `npx source-map-explorer build/static/js/main.eaa15ea4.chunk.js`
- Verify the `@carbon/icons-react` package does not include the full 1MB of icon assets

![image](https://user-images.githubusercontent.com/3901764/61565926-25c7f100-aa40-11e9-9faa-0ff314ab34fc.png)
